server/chat-plugins/sample-teams.ts

```diff
@@ -0,0 +1,515 @@
+/**
+ * A sample teams plugin
+ *
+ * @author Kris
+ */
+
+import {FS, Utils} from "../../lib";
+
+const SAMPLE_TEAMS = 'config/chat-plugins/sample-teams.json';
+
+interface SampleTeamsFile {
+	whitelist: {[formatid: string]: RoomID[]};
+	/** Teams are stored in the packed format */
+	teams: {
+		[formatid: string]: {
+			uncategorized: {[k: string]: string},
+			[category: string]: {[teamName: string]: string},
+		},
+	};
+}
+
+const file: SampleTeamsFile = JSON.parse(FS(SAMPLE_TEAMS).readIfExistsSync() || `{"whitelist": {}, "teams": {}}`);
+
+function save() {
+	FS(SAMPLE_TEAMS).writeUpdate(() => JSON.stringify(file));
+}
+
+if (!file.whitelist) file.whitelist = {};
+if (!file.teams) file.teams = {};
+for (const formatid in file.teams) {
+	if (!file.teams[formatid].uncategorized) file.teams[formatid].uncategorized = {};
+}
+save();
+
+const SampleTeams = new class SampleTeams {
+	isRoomStaff(user: User, roomids: RoomID[]) {
+		let matched = false;
+		if (!roomids?.length) return false;
+		for (const roomid of roomids) {
+			const room = Rooms.search(roomid);
+			// Malformed entry from botched room rename
+			if (!room) throw new Error(`Room ${roomid} not found`);
+			matched = room.auth.isStaff(user.id);
+			if (matched) break;
+		}
+		return matched;
+	}
+
+	checkPermissions(user: User, roomids: RoomID[]) {
+		// Give Development room mods access to help fix crashes
+		return this.isRoomStaff(user, roomids) || user.can('bypassall') || !!Rooms.get('development')?.auth.atLeast(user, '@');
```

Comment:
The Rooms.get('development').. etc call is kinda messy, maybe move the dev mod check into a new function?